mm2src/coins/z_coin/storage/blockdb/blockdb_idb_storage.rs

```diff
@@ -0,0 +1,252 @@
+use crate::z_coin::storage::validate_chain;
+use crate::z_coin::storage::{BlockDbImpl, BlockProcessingMode, CompactBlockRow, ZcoinConsensusParams};
+use crate::z_coin::z_coin_errors::ZcoinStorageError;
+
+use async_trait::async_trait;
+use mm2_core::mm_ctx::MmArc;
+use mm2_db::indexed_db::{BeBigUint, DbIdentifier, DbInstance, DbUpgrader, IndexedDb, IndexedDbBuilder, InitDbResult,
+                         MultiIndex, OnUpgradeResult, TableSignature};
+use mm2_db::indexed_db::{ConstructibleDb, DbLocked};
+use mm2_err_handle::prelude::*;
+use num_traits::ToPrimitive;
+use protobuf::Message;
+use std::path::Path;
+use zcash_client_backend::proto::compact_formats::CompactBlock;
+use zcash_extras::WalletRead;
+use zcash_primitives::block::BlockHash;
+use zcash_primitives::consensus::BlockHeight;
+
+const DB_NAME: &str = "z_compactblocks_cache";
+const DB_VERSION: u32 = 1;
+
+pub type BlockDbRes<T> = MmResult<T, ZcoinStorageError>;
+pub type BlockDbInnerLocked<'a> = DbLocked<'a, BlockDbInner>;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct BlockDbTable {
+    height: u32,
+    data: Vec<u8>,
+    ticker: String,
+}
+
+impl BlockDbTable {
+    pub const TICKER_HEIGHT_INDEX: &str = "block_height_ticker_index";
+}
+
+impl TableSignature for BlockDbTable {
+    fn table_name() -> &'static str { "compactblocks" }
```

Comment:
Updating `TableSignature` trait would be better in favor of replacing `fn table_name() -> &static str` to `const TABLE_NAME: &str';`

Reply:
Yes, as it is quite easy change(just replacing bunch of lines)

Reply:
done